frame/uniques/src/lib.rs

```diff
@@ -92,7 +94,14 @@ pub mod pallet {
 		type Event: From<Event<Self, I>> + IsType<<Self as frame_system::Config>::Event>;
 
 		/// Identifier for the collection of item.
-		type CollectionId: Member + Parameter + MaxEncodedLen + Copy;
+		type CollectionId: Member
+			+ Parameter
+			+ MaxEncodedLen
+			+ Copy
+			+ Default
+			+ Unsigned
+			+ FromPrimitive
+			+ Saturating;
```

Comment:
```suggestion
		type CollectionId: Member
			+ Parameter
			+ AtLeast32BitUnsigned
			+ MaxEncodedLen
			+ Copy
			+ Default;
```

I think we should stick to Substrate traits as to not confuse developers.  
This should do everything you need, so you can remove the `num-traits` dependency.  

Instead of `T::CollectionId::from_u8(1).unwrap()` you can then use `1u32.into()` as  well.